src/wallet.js

```diff
@@ -1,142 +1,141 @@
-                        ])
-                    tx.addOutput(ret, 0);
-                }
-                if (Total < amount && process.env.NODE_ENV != undefined) {
-                    reject("Not enough coin to pay total " + Total + "  " + amount);
-                    return;
-                }
-                logger.log('debug', "Total: " + Total);
-                logger.log('debug', "Amount: " + amount);
-                payees.forEach((payee) => {
-                    logger.log('debug', payee);
-                    tx.addOutput(payee.wallet, payee.btcamount);
-                });
-                calcFee(tx).then((fee) => {
-                    tx.addOutput(payor.address, Number(Total - (amount + fee)));
-                    logger.log('debug', "Fee: " + fee);
-                    logger.log('info', 'Sending back %d', Number(Total - amount - fee));
-                    tx.sign(0, key);
-                    logger.log("info", tx);
-                    const lengthTransaction = tx.build().toHex().length / 2;
-                    console.log("Transaction size: ", lengthTransaction);
-                    if (process.env.NODE_ENV == undefined) {  // if there is no NODE_ENV its in production and actually send
-                        console.log("Sending");
-                        pushtx.pushtx(tx.build().toHex(), null).then((result) => {
-                            fulfill(amount)
-                        }).catch(function (err) {
-                            reject(err);
-                        });
-                    }
-                    else {
-                        logger.log('info', 'Dev environment not sending');
-                        fulfill(amount);
-                    }
-                })
-
-            });
-        })
+        //   console.log(data.txs[q]);
+    }
+
+    if (opcode) {
+        const ret = bitcoin.script.compile([
+            bitcoin.opcodes.OP_RETURN,
+            data
+        ]);
+        tx.addOutput(ret, 0);
+    }
+
+    if (Total < amount && process.env.NODE_ENV != undefined) {
+        return Promise.reject('Not enough coin to pay total ' + Total + '  ' + amount);
+    }
+
+    logger.log('debug', 'Total: ' + Total);
+    logger.log('debug', 'Amount: ' + amount);
+
+    payees.forEach((payee) => {
+        logger.log('debug', payee);
+        tx.addOutput(payee.wallet, payee.btcamount);
     });
-}
-let getAddress = () => {
+
+    const fee = await calcFee(tx);
+
+    tx.addOutput(payor.address, Number(Total - (amount + fee)));
+    tx.sign(0, key);
+
+    logger.log('debug', 'Fee: ' + fee);
+    logger.log('info', 'Sending back %d', Number(Total - amount - fee));
+    logger.log('info', tx);
+
+    const lengthTransaction = tx.build().toHex().length / 2;
+    console.log('Transaction size: ', lengthTransaction);
+
+    if (process.env.NODE_ENV === undefined || process.env.NODE_ENV === 'production') {  // if there is no NODE_ENV its in production and actually send
```

Comment:
added `process.env.NODE_ENV === 'production'` since it's very common to have it like that in production environments.